doc/specs/stdlib_sorting.md

```diff
@@ -0,0 +1,644 @@
+algorithms on random data. On partially sorted data it can show either
+slower `heap sort` performance, or enhanced performance by up to a
+factor of six. Still, even when it shows enhanced performance, its
+performance on partially sorted data is typically an order of
+magnitude slower than `ORD_SORT`. Its memory requirements are also
+low, being of order O(Ln(N)), while the memory requirements of
+`ORD_SORT` and `SORT_INDEX` are of order O(N).
+
+### Tentative specifications of the `stdlib_sorting` procedures
+
+#### `ord_sort` - sorts an input array
+
+##### Status
+
+Experimental
+
+##### Description
+
+Returns an input `array` with the elements sorted in order of
+increasing value.
+
+##### Syntax
+
+`call [[stdlib_sorting(module):ord_sort(subroutine)]]ord_sort ( array[, work ] )`
+
+##### Class
+
+Generic subroutine.
+
+##### Arguments
+
+`array` : shall be a rank one array of any of the types:
+`integer(int8)`, `integer(int16)`, `integer(int32)`, `integer(int64)`,
+`real(sp)`, `real(dp)`, `real(qp)`, `character(*)`, or
+`type(string_type)`. It is an `intent(inout)` argument. On input it is
+the array to be sorted. If both the type of `array` is real and at
+least one of the elements is a `NaN`, then the ordering of the result
+is undefined. Otherwise on return its elements will be sorted in order
+of non-decreasing value.
+
+`work` (optional): shall be a rank one array of the same type as
+array, and shall have at least `size(array)/2` elements. It is an
+`intent(inout)` argument. It is intended to be used as "scratch"
+memory for internal record keeping. If associated with an array in
+static storage, its use can significantly reduce the stack memory
+requirements for the code. Its contents on return are undefined.
+
+##### Notes
+
+`ORD_SORT` implements a hybrid sorting algorithm combining
+`merge sort`, and `insertion sort`. For most purposes it behaves like
+a `merge sort`, providing worst case `O(N Ln(N))` run time performance
+for most random arrays, that is typically slower than `SORT`.
+However, if the array has significant runs of decreasing or
+non-decreasing values, performance can be much better than
+`SORT`, with `O(N)` behavior on uniformly decreasing, or
+non-decreasing arrays. The optional `work` array replaces "scratch"
+memory that would otherwise be allocated on the stack. If `array` is of
+any type `REAL` the order of its elements on return undefined if any
+element of `array` is a `NaN`.  Sorting of `CHARACTER(*)` and
+`STRING_TYPE` arrays are based on the operator `>`, and not on the
+function `LGT`.
+
+
+##### Example
+
+```fortran
+    ...
+    ! Read arrays from sorted files
+    call read_sorted_file( 'dummy_file1', array1 )
+    call read_sorted_file( 'dummy_file2', array2 )
+    ! Concatenate the arrays
+	array = [ array1, array2 ]
+    ! Sort the resulting array
+    call ord_sort( array, work )
+    ! Process the sorted array
+    call array_search( array, values )
+    ...
+```
```

Comment:
````suggestion
```fortran
program demo_ord_sort
  use stdlib_sorting, only: ord_sort
  implicit none
  integer, allocatable :: array1(:), work(:)
  
  array1 = [ 5, 4, 3, 1, 10, 4, 9]
  allocate(work, mold = array1)
  call ord_sort(array1, work)
  print*, array1   !print [1, 3, 4, 4, 5, 9, 10]
end program demo_ord_sort
```
````
suggestion: a small program that could be copyied/pasted, and compiled

Reply:
Done